web/js/browser-support.js

```diff
@@ -0,0 +1,78 @@
+function webAssemblySupported() {
+	return "WebAssembly" in window
+		&& typeof window.WebAssembly.instantiate === "function";
+}
+
+function serviceWorkersSupported() {
+	return "serviceWorker" in navigator;
+}
+
+function webWorkersSupported() {
+	return "Worker" in window;
+}
+
+function indexedDbSupported() {
+	return "indexedDB" in window
+		&& typeof window.indexedDB.open === "function";
+}
+
+function sessionStorageSupported() {
+	var localStorageAvailable = "localStorage" in window
```

Comment:
AFAIK, no browsers have one and not the other, but... in our case, we only ever plan to use `sessionStorage`, so I think we can safely skip the test for `localStorage`.